frame/support/src/storage/generator/map.rs

```diff
@@ -170,19 +169,26 @@ impl<
 			match sp_io::storage::next_key(&previous_key).filter(|n| n.starts_with(&prefix)) {
 				Some(next) => {
 					previous_key = next;
-					let maybe_value = unhashed::get::<O>(&previous_key);
-					match maybe_value {
-						Some(value) => {
-							let mut key_material = G::Hasher::reverse(&previous_key[prefix.len()..]);
-							match K::decode(&mut key_material) {
-								Ok(key) => match f(key, value) {
-									Some(new) => unhashed::put::<V>(&previous_key, &new),
-									None => unhashed::kill(&previous_key),
-								},
-								Err(_) => continue,
-							}
-						}
-						None => continue,
```

Comment:
note the behavior here hasn't changed, but I added some log and I think it improves readability